virtual_ecosystem/models/hydrology/hydrology_model.py

```diff
@@ -419,13 +418,15 @@ def _update(self, time_index: int, **kwargs: Any) -> None:
         and a number of parameters that as described in detail in
         :class:`~virtual_ecosystem.models.hydrology.constants.HydroConsts`.
         """
-        # Determine number of days, currently only 30 days (=1 month)
-        if self.model_timing.update_interval_quantity != Quantity("1 month"):
-            to_raise = NotImplementedError("This time step is currently not supported.")
-            LOGGER.error(to_raise)
-            raise to_raise
-
-        days: int = 30
+        # Determine number of days
+        days_float: float = self.model_timing.update_interval_seconds / 86400
+        days: int = int(days_float // 1)
+
+        # Check if the number of days is exact and warn if not
+        if not np.allclose(days_float % 1, 0):
+            LOGGER.warning(
+                "Update interval is not a whole number of days ({n_days}) using floor."
```

Comment:
What is floor?

Reply:
Basically round down to nearest integer. The code used to use the `floor` function. I'll clarify.